spec/models/dependent/rules_spec.rb

```diff
@@ -0,0 +1,265 @@
+      context "with a dependent who has a qualified child relationship but doesn't meet age conditions" do
+        let(:birth_date) { Date.new(tax_year - 70, 12, 25) }
+        let(:qualifying_child_relationship) { true }
+
+        it "returns true" do
+          expect(subject.qualifying_relative?).to eq true
+        end
+
+        context "when meet misc requirements are not met" do
+          let(:meets_misc_qualifying_relative_requirements) { false }
+
+          it "returns false" do
+            expect(subject.qualifying_relative?).to eq false
+          end
+        end
+      end
+
+      context "with a dependent who has a qualified relative relationship" do
+        let(:birth_date) { Date.new(tax_year - 30, 12, 25) }
+        let(:qualifying_relative_relationship) { true }
+
+        it "returns true" do
+          expect(subject.qualifying_relative?).to eq true
+        end
+
+        context "when meet misc requirements are not met" do
+          let(:meets_misc_qualifying_relative_requirements) { false }
+
+          it "returns false" do
+            expect(subject.qualifying_relative?).to eq false
+          end
+        end
+      end
+    end
+  end
+
+  describe ".meets_qc_residence_condition?" do
+    context "when already generally meeting the qualified child residence conditions" do
+      let(:meets_qc_residence_condition_generic) { true }
+
+      context "when younger than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 12, 25) }
+        it "returns true" do
+          expect(subject.meets_qc_residence_condition?).to eq(true)
+        end
+      end
+
+      context "when older than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 1, 1) }
+        it "returns true" do
+          expect(subject.meets_qc_residence_condition?).to eq(true)
+        end
+      end
+    end
+
+    context "when not generally meeting the qualified child residence conditions" do
+      let(:meets_qc_residence_condition_generic) { false }
+
+      context "when younger than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 12, 25) }
+        it "returns true" do
+          expect(subject.meets_qc_residence_condition?).to eq(true)
+        end
+      end
+
+      context "when older than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 1, 1) }
+        it "returns false" do
+          expect(subject.meets_qc_residence_condition?).to eq(false)
+        end
+      end
+    end
+  end
+
+  describe ".qualifying_child?" do
+    context "as a child who generally qualifies" do
+      let(:qualifying_child_relationship) { true }
+      let(:ssn_present) { true }
+      let(:meets_qc_claimant_condition) { true }
+      let(:meets_qc_misc_conditions) { true }
+
+      context "when the child is < 6 months old" do
+        let(:birth_date) { Date.new(tax_year, 12, 25) }
+
+        it "returns true" do
+          expect(subject.qualifying_child?).to eq(true)
+        end
+
+        context "and not born yet" do
```

Comment:
i guess we could remove this now that the < 0 age condition is rolled into the age condition method